internal/k8s/controllers/reconciliation_test.go

```diff
@@ -426,3 +445,34 @@ var _ = Describe("FRRK8S Controller", func() {
 		})
 	})
 })
+
+var _ = Describe("Layer2 Status Controller", func() {
+	Context("SetupWithManager", func() {
+		It("Should Reconcile correctly", func() {
+			name := "test-service"
+
+			// simulate some service is advertised
+			layer2ServiceAdvs = append(layer2ServiceAdvs, layer2.NewIPAdvertisement(net.IP("127.0.0.1"), true, make(sets.Set[string])))
+			// notify reconciler to reconcile
+			layer2StatusUpdateChan <- NewL2StatusEvent(testNamespace, name)
+			Eventually(func() string {
+				toCheck := v1beta1.ServiceL2Status{}
+				err := k8sClient.Get(context.TODO(), client.ObjectKey{Name: name, Namespace: testNamespace}, &toCheck)
+				if err != nil {
+					return ""
+				}
+				return toCheck.Status.Node
+			}, 5*time.Second, 200*time.Millisecond).Should(Equal(testNodeName))
+
```

Comment:
can you add an update to the interface and verify they are reflected in the status?